src/components/MessageList.tsx

```diff
@@ -0,0 +1,486 @@
+
+import React, { useState, useEffect, useRef } from 'react';
+import { observer } from 'mobx-react-lite';
+import {
+	Droppable,
+	DroppableProvided,
+	DropResult,
+	DragDropContext,
+	Draggable,
+	DraggableProvided,
+	DraggableStateSnapshot,
+} from 'react-beautiful-dnd';
+import { nanoid } from '../../node_modules/nanoid';
+import { useStore } from '../hooks/useStore';
+import '../styles/message-list.scss';
+import '../styles/splitter.scss';
+import SplitView from '../split-view/SplitView';
+import SplitViewPane from '../split-view/SplitViewPane';
+import { downloadFile } from '../helpers/downloadFile';
+import {
+	ParsedMessageItem,
+	ActMessageItem,
+	isParsedMessageItem,
+	isActMessageItem,
+} from '../models/Message';
+
+export type Indicator =
+	| 'indicator-unvisible'
+	| 'indicator-edited'
+	| 'indicator-successful'
+	| 'indicator-unsuccessful';
+
+interface EditMessageProps {
+	editMessageMode: boolean;
+	editedMessageIndex: number;
+	indicators: Indicator[];
+}
+
+const Messages = () => {
+	const store = useStore();
+	const [isReplay, setReplayMode] = useState(false);
+	const isReplayRef = useRef(isReplay);
+
+	const loadFromFile = (file: FileList | null) => {
+		if (file != null) {
+			const reader = new FileReader();
+			reader.readAsText(file.item(0) as Blob);
+			reader.onload = () => {
+				if (typeof reader.result === 'string') {
+					jsonMessagesFromString(reader.result);
+				}
+			};
+		}
+	};
+
+	const jsonMessagesFromString = (rawFromFile: string) => {
+		try {
+			const json = JSON.parse(rawFromFile);
+			store.clearParsedMessages();
+
+			for (let i = 0; i < json.length; i++) {
+				store.addParsedMessage(json[i]);
+			}
+		} catch (error) {
+			// eslint-disable-next-line no-alert
+			alert('Failed to read the file. Please, try to select another file');
+		}
+	};
+
+	useEffect(() => {
+		isReplayRef.current = isReplay;
+		if (isReplay) {
+			store.setEditMessageMode(false);
+			replaySendMessage(store.getCurrentMessagesArray, 0);
+		}
+	}, [isReplay]);
+
+	const replaySendMessage = (array: ParsedMessageItem[] | ActMessageItem[], index: number) => {
+		if (isReplayRef.current && array.length > 0 && index < array.length) {
+			setTimeout(() => {
+				store.replayMessage(array[index], index).then(() => {
+					if (index === array.length - 1) {
+						setReplayMode(false);
+					} else {
+						replaySendMessage(array, index + 1);
+					}
+				});
+			}, array[index].delay);
+		}
+	};
+
+	const exportFn = () => {
+		downloadFile(
+			JSON.stringify(store.getCurrentMessagesArray),
+			store.selectedSchemaType === 'parsed-message' ? 'parsedMessages' : 'actMessages',
+			'application/json',
+		);
+	};
+
+	return (
+		<div>
+			<MessageEditArea
+				messages={store.getCurrentMessagesArray}
+				indicators={store.indicators.slice()}
+				editMessageMode={store.editMessageMode}
+				editedMessageIndex={store.editedMessageIndex}
+				messageListPanelArea={store.messageListPanelArea}
```

Comment:
panelarea can be stored and handled inside MessageList no need to keep it in global store